packages/core/streams/streaming-text-response.ts

```diff
@@ -16,29 +16,116 @@ export class StreamingTextResponse extends Response {
   }
 }
 
+/**
+ * A simple utility function to convert a ReadableStream to a Node.js Readable.
+ * This is campatible with Node.js 18 web streams and polyfill-stream package both.
+ */
+export function streamToNodeReadable(stream: ReadableStream) {
+  const reader = stream.getReader()
+  let closed = false
+
+  // in order to avoid module check error
+  const Readable = require('stream')
+    .Readable as typeof import('stream').Readable
+
+  const readable = new Readable({
+    read() {
+      reader
+        .read()
+        .then(({ done, value }) => {
+          if (done) {
+            this.push(null)
+          } else {
+            this.push(value)
+          }
+        })
+        .catch(e => {
+          readable.destroy(e)
+        })
+    },
+    destroy(error, callback) {
+      const done = () => {
+        try {
+          callback(error)
+        } catch (err: unknown) {
+          process.nextTick(() => {
+            throw err
+          })
+        }
+      }
+
+      if (!closed) {
+        reader.cancel(error).then(done, done)
+        return
+      }
+      done()
+    }
+  })
+
+  reader.closed.then(
+    () => {
+      closed = true
+    },
+    error => {
+      readable.destroy(error)
+    }
+  )
+
+  return readable
+}
+
+/**
+ * A utility function to get a OutgoingHttpHeaders from a HeadersInit.
+ */
+export function headersInitToOutgoingHeaders(headers: HeadersInit | undefined) {
+  const h = new Headers(headers)
+  const outgoingHeaders: OutgoingHttpHeaders = {}
+  h.forEach((value, key) => {
+    if (
+      key === 'set-cookie' &&
+      'getSetCookie' in h &&
+      typeof h.getSetCookie === 'function'
+    ) {
+      // nodejs 18.14.1 supports it, otherwise developer should polyfill this method for their own
+      outgoingHeaders[key] = h.getSetCookie()
+      return
+    }
```

Comment:
I don't really like this special handling because it doesn't fail when `getSetCookie` isn't defined, and the fallback has subtly incorrect behavior. We can implement this with the same amount of code with:

```js
    const old = outgoingHeaders[key]
    if (Array.isArray(old)) {
      old.push(value)
    } else {
      outgoingHeaders[key] = old === undefined ? value : [String(old), value]
    }
```

Reply:
For users who have already polyfilled getSetCookie, this is OK, and if not， the fallback follows the original implementations which means users dont care.

Last you change headers set to headers append, which I think is overthought. In `Response`, there is only set-cookie key which really needs appending. Others like content-type should be only reserved once. Moreover content-encoding should be omitted, cause we already decode the content in the response reader.